source/sessions/snapshot-sessions.rst

```diff
@@ -0,0 +1,351 @@
+server have been overloaded to take a session parameter. If that session was
+started with ``isSnapshot = true`` then all operations using that
+session will share the same snapshot.
+
+ReadConcern changes
+===================
+
+``snapshot`` added to `ReadConcernLevel enumeration <../read-write-concern/read-write-concern.rst#read-concern>`_.`.
+
+Server Commands
+===============
+
+There are no new server commands related to snapshot reads. Instead,
+snapshot reads are implemented by:
+
+1. Saving the ``atClusterTime`` returned by 5.0+ servers for the first find/aggregate operation in a
+   property ``snapshotTimestamp`` of the ``ClientSession`` object. Drivers MUST save the ``atClusterTime``
+   in the ``ClientSession`` object.
+
+2. Passing that ``snapshotTimestamp`` in the ``atClusterTime`` field of the ``readConcern`` field
+   for subsequent snapshot read operations (for find/aggregate/distinct commands).
+
+Server Command Responses
+========================
+
+To support snapshot reads the server returns the ``atClusterTime`` in
+cursor object it sends to the driver (for both find/aggregate commands).
+
+.. code:: typescript
+
+    {
+        ok : 1 or 0,
+        ... // the rest of the command reply
+        cursor : {
+            ... // the rest of the cursor reply
+            atClusterTime : <BsonTimestamp>
+        }
+    }
+
+The ``atClusterTime`` MUST be stored in the ``ClientSession`` to later be passed as the
+``atClusterTime`` field of the ``readConcern`` with ``snapshot`` level field  in subsequent read operations.
+
+Server Errors
+=============
+1. The server may reply to read commands with a ``SnapshotTooOld`` error if the client's ``atClusterTime`` value is not available in the server's history.
+2. The server will return ``InvalidOptions`` error if both ``atClusterTime`` and ``afterClusterTime`` options are set to true.
+
+Snapshot read commands
+======================
+
+For snapshot reads the driver MUST first obtain ``atClusterTime`` from cursor response of find/aggregate command,
+by specifying ``readConcern`` with ``snapshot`` level field, and store it as ``snapshotTimestamp`` in 
+``ClientSession`` object.
+
+.. code:: typescript
+
+    {
+        find : <string>, // or other read command
+        ... // the rest of the command parameters
+        readConcern :
+        {
+            level : "snapshot"
+        }
+    }
+
+For subsequent reads from same snapshot driver MUST send the ``snapshotTimestamp`` saved in
+the ``ClientSession`` as the value of the ``atClusterTime`` field of the
+``readConcern`` with ``snapshot`` level field:
+
+.. code:: typescript
+
+    {
+        find : <string>, // or other read command
+        ... // the rest of the command parameters
+        readConcern :
+        {
+            level : "snapshot",
+            afterClusterTime : <BsonTimestamp>
+        }
+    }
+
+Lists of commands that support snapshot reads:
+
+1. find
+2. aggregate
+3. distinct
+
+Snapshot read commands errors
```

Comment:
I'm not convinced its necessary to specify this, since retryable reads are only enabled via an allow-list anyway.

Reply:
Thanks, removed.